File: libs/FTBQuests.Validation/ValidationIssue.cs
// <copyright file="ValidationIssue.cs" company="CyberDay1">
// Copyright (c) CyberDay1. All rights reserved.
// </copyright>

namespace FTBQuests.Validation;

/// <summary>
/// Represents a validation issue discovered while analyzing a quest pack.
/// </summary>
/// <param name="Severity">The severity of the issue.</param>
/// <param name="Path">The dotted JSON path to the failing value.</param>
/// <param name="Message">Human-readable description of the issue.</param>
/// <param name="Code">Stable machine-readable code describing the issue.</param>
public sealed record ValidationIssue(ValidationSeverity Severity, string Path, string Message, string Code);

File: core-domain/src/main/java/dev/ftbq/editor/validation/ValidationIssue.java
package dev.ftbq.editor.validation;

import java.util.Objects;

public record ValidationIssue(String severity, String path, String message) {
    public ValidationIssue {
        Objects.requireNonNull(severity, "severity");
        Objects.requireNonNull(path, "path");
        Objects.requireNonNull(message, "message");
    }
}
